js/piwik.js

```diff
@@ -3364,10 +3356,7 @@ if (typeof window.Piwik !== 'object') {
 
                 var cookieValue = visitorIdCookieValues.uuid + '.' +
                     visitorIdCookieValues.createTs + '.' +
-                    visitorIdCookieValues.visitCount + '.' +
```

Comment:
@diosmosis should we maybe keep BC for existing cookies? Currently, any existing cookie might break as they might have different order eg for the nowTs ?

Reply:
I haven't reviewed the code in detail, but I'm not clear on how we can calculate Visit Count, without storing it in the cookie. I'm probably missing something but how is it possible to keep a count of "visit count" without storing the count in the cookie? (Since our "Window look back" is only 30min by default, we wouldn't be able to lookup past visits)

Reply:
btw we don't just look back 30min AFAIK at least not when visitorId is used. And when configId only is used then we can basically assume anyway it is the first visit. 

visitcount is maybe also quite inaccurate in some cases but mostly works.

Reply:
> (Since our "Window look back" is only 30min by default, we wouldn't be able to lookup past visits)

this is only for fingerprint matching, not for matching based on idvisitor

Reply:
> (Since our "Window look back" is only 30min by default, we wouldn't be able to lookup past visits)

this is only for matching by config_id. and since the JS tracker sends a visitor ID, and no other tracker sends the visit count cookie automatically, I don't think this behavior is affected.

Reply:
i can test this on demo2 via log replay to make sure the data is still accurate

Reply:
@diosmosis just checking whether you checked regarding https://github.com/matomo-org/matomo/pull/15774/files#r423417650 because now `nowTs` is in third position vs before existing cookies would have stored the visitCount there? Not sure if we need to check in `getValuesFromVisitorIdCookie` whether the value looks more like a timestamp?

Also I reckon in `getValuesFromVisitorIdCookie()` the array indexes 4 and 6 might not always be defined?
![image](https://user-images.githubusercontent.com/273120/82858289-8df03900-9f67-11ea-9c04-a4f57744d678.png)

The `currentVisitTs` seems no longer used in general?